catroid/src/org/catrobat/catroid/io/StorageHandler.java

```diff
@@ -393,6 +395,14 @@ public boolean saveProject(Project project) {
 			writer = new BufferedWriter(new FileWriter(tmpCodeFile), Constants.BUFFER_8K);
 			writer.write(projectXml);
 			writer.flush();
+
+			FileWriter permissionWriter = new FileWriter(buildProjectPath(project.getName()) + PROJECTPERMISSIONS_NAME);
+
+			for(String str: project.getRequiredResources()) {
```

Comment:
`str` to something that isn't an abbreviation, like `resource` would be my guess.